mozilla/perfcompare
# moved framework dropdown to results table

This PR closes the following issue: https://mozilla-hub.atlassian.net/browse/PCF-421 

This PR is broken down into 3 commits:

- [x] [ Create a separate framework component for the Results Page table.](https://github.com/mozilla/perfcompare/pull/691/commits/a20f066a8df76f03e8d71abf40170b2019979d3d)
         I didn't want to force the new design and functionality unto the current `framework` component in `Search` so I created a new component in `Results` to reduce clutter, separate concerns, and easily manage the comps.

<img width="1221" alt="Screenshot 2024-07-23 at 12 20 37" src="https://github.com/user-attachments/assets/4bf60fb1-cefb-4ecc-a69e-b09872c4cd51">

- [x] [Use `useSearchParams` to change the url, refetch the data and rerender the page ](https://github.com/mozilla/perfcompare/pull/691/commits/25728d011392ed89d81d1d467d8391a3e742c4bf)
       I also added a _hidden input_ to keep the framework data in the url when the users change the revisions via the 'Edit entry' button. 

- [x] [Add test coverage](https://github.com/mozilla/perfcompare/pull/691/commits/1d788e3da8caaec935b31f770218a12dd96cefa1)

To Test:
1.) Open the PR's deployed link: https://deploy-preview-691--mozilla-perfcompare.netlify.app/
2.) Select a base revision and new revision(s) in the `Compare with a base` component
3.) After Results page has finished loading, change the framework in the menu bar above the Results table.
4. ) Spinner should appear as the table loads and disappears when finished loading new results.
5. ) Click `Edit entry` and change one of the revisions.
6. ) Press `Compare`
7. ) The framework value should not change and is not missing in the url. 
8. ) Repeat with `Compare over time`
